src/iap/shared/services/jwt-auth.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core';
+
+import * as dayjs from 'dayjs';
+import { Observable, map, shareReplay, tap } from 'rxjs';
+
+import { apiRoutesAuth } from '@shared/constants/api-routes';
+import { EXPIRES_AT, ID_TOKEN } from '@shared/constants/auth';
+import { JwtSession } from '@shared/models/dto/jwt-session.dto';
+import { User } from '@shared/models/entities/user.entity';
+import { ApiService } from '@shared/services/api.service';
+
+// TODO: add refresh token to this strategy
+@Injectable({ providedIn: 'root' })
+export class JwtAuthService {
+	constructor(private api: ApiService) {}
+
+	login(login: string, password: string): Observable<JwtSession> {
+		const url = apiRoutesAuth.login;
+		return this.api.post(url, { login, password }).pipe(
+			tap((res: any) => this.setSession(res as JwtSession)),
+			shareReplay(),
```

Comment:
Use your new possibilities!
```suggestion
		return this.api.post<JwtSession>(url, { login, password }).pipe(
			tap((res) => this.setSession(res)),
			shareReplay(),
```